### PR TITLE
Add  io.github.martinrotter.textosaurus

### PR DESCRIPTION
**Pull request by martinrotter**
https://github.com/martinrotter/textosaurus

### CODE REVIEW

**Comment by TingPing** [comment]
The app-id should be `io.github.martinrotter.textosaurus`

File: com.github.textosaurus.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+                     "--filesystem=xdg-download",
+                     "--talk-name=org.kde.StatusNotifierWatcher",
+                     "--talk-name=org.freedesktop.Notifications" ],
+    "modules": [
+        {
+            "name": "textosaurus",
+            "buildsystem": "qmake",
+            "post-install": [
+                "install -d /app/share/icons/hicolor/512x512/apps",
+                "install -m644 /app/share/pixmaps/textosaurus.png /app/share/icons/hicolor/512x512/apps/textosaurus.png"
```
Comment (TingPing) [inline com.github.textosaurus.json]: Pass `-D` here and you can remove the previous line. You will also want to delete the file from `pixmaps` and honestly just change that upstream as nobody should be using that directory.

**Comment by TingPing** [comment]
Review of the appdata file:

- `launchable` refers to wrong desktop file
- missing `screenshots`
- missing `releases`

File: com.github.textosaurus.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    "runtime-version": "5.9",
+    "sdk": "org.kde.Sdk",
+    "command": "textosaurus",
+    "rename-icon": "textosaurus",
+    "finish-args": [ "--device=dri",
+                     "--share=ipc",
+                     "--share=network",
+                     "--socket=x11",
+                     "--socket=wayland",
+                     "--filesystem=xdg-download",
```
Comment (TingPing) [inline com.github.textosaurus.json]: Surely as a text editor you want more access than that.

**Comment by martinrotter** [comment]
> The app-id should be io.github.martinrotter.textosaurus

Why? If I take a look into flathub org github, I see many `com.github.xxxx` apps.

File: com.github.textosaurus.json (hunk 2 above)
Comment (martinrotter) [reply com.github.textosaurus.json]: Sure.

File: com.github.textosaurus.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+    "build-options": {
```
Comment (TingPing) [inline com.github.textosaurus.json]: This is a very strange layout. Please be consistent with the order other manifest use.

Comment (martinrotter) [reply com.github.textosaurus.json]: Is order somehow defined in Flatpak docs? I can reorder it I guess, just autoformatted.

File: com.github.textosaurus.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        {
+            "build-commands": [
+                "mkdir -p /app",
+                "$CXX --version",
+                "qmake -v",
+                "qmake textosaurus.pro QMAKE_CXX=$CXX -r CONFIG+=release PREFIX=/app",
+                "make -j3",
+                "make install"
+            ],
+            "buildsystem": "simple",
```
Comment (TingPing) [inline com.github.textosaurus.json]: As I said before, `simple` isn't needed. Use `qmake` and set `config-opts`.

Comment (martinrotter) [reply com.github.textosaurus.json]: Is there somewhere docs for qmake buildsystem?

**Comment by martinrotter** [comment]
>All of them contain, or should, the org name. Yours does not. io vs com is less consistent but I think it makes sense to reserve the former for user repos and the latter for official Github projects.
So should I add "io" or stay with "com" and add org name?

File: com.github.textosaurus.json (hunk 3 above)
Comment (TingPing) [reply com.github.textosaurus.json]: No it isn't defined. Alphabetical just doesn't make sense.

File: com.github.textosaurus.json (hunk 4 above)
Comment (TingPing) [reply com.github.textosaurus.json]: Not specifically. Here is an example:

```json
  {
    "buildsystem": "qmake",
    "config-opts": [
      "CONFIG+=release"
    ]
   }
```

**Comment by TingPing** [comment]
I think `io.github.martinrotter.textosaurus` is the best choice.

**Comment by martinrotter** [comment]
Will do.

**Comment by martinrotter** [comment]
@TingPing Kindle requesting review, fixed all suggested changes.

**Comment by martinrotter** [comment]
+ fixed paths for flatpak-specific version of Textosaurus according to http://docs.flatpak.org/en/latest/conventions.html#xdg-base-directories, moved icon to hicolor/....

**Comment by TingPing** [comment]
bot, build io.github.martinrotter.textosaurus

File: io.github.martinrotter.textosaurus.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+            "CXX": "/usr/lib/sdk/gcc7/bin/g++"
+        }
+    },
+    "modules": [
+        {
+            "config-opts": [
+                "QMAKE_CXX=/usr/lib/sdk/gcc7/bin/g++",
+                "CONFIG+=release",
+                "CONFIG+=FLATPAK_MODE",
+                "PREFIX=/app"
```
Comment (TingPing) [inline io.github.martinrotter.textosaurus.json]: Prefix is already set for you.

**Comment by TingPing** [comment]
> fixed paths for flatpak-specific version of Textosaurus according to http://docs.flatpak.org/en/latest/conventions.html#xdg-base-directories

That looks like a misunderstanding. You shouldn't change anything and the previous behavior looked fine.

**Comment by martinrotter** [comment]
"config" folder for application when run under Flatpak is "sandboxed" for the one single app (if I get it correctly), now, the source code **literally** follows Flatpak docs, which says `~/.var/app/<app-id>/config` so I coded/fixed the app to use that path literally = `~/.var/app/io.github.martinrotter.textosaurus/config/config.ini`

**Comment by TingPing** [comment]
> @TingPing Check the screenshot, it is taken from app when running from flatpak-builder.

Yes it is wrong and the other codepath was right.

**Comment by martinrotter** [comment]
@TingPing Then your docs say it just wrong. But I can revert it, just to prove that I am kind.

**Comment by martinrotter** [comment]
@TingPing Fixed it, same textosaurus tag/release, also added OARS to appdata.

**Comment by TingPing** [comment]
Anyway everything built fine just update it to the latest commit and ensure you are happy with the results using it.

**Comment by martinrotter** [comment]
@TingPing  Wait, what do you mean with " update it to the latest commit"?

**Comment by TingPing** [comment]
Oh have you been updating the `2018.04.16` tag? If so please don't do that and also include 
`"commit": "15dcb218150efe2e21de6acaa9614c48c18fa860"` so it is clear what is building.

**Comment by martinrotter** [comment]
Yes, I updated the tag.

**Comment by martinrotter** [comment]
@TingPing  done

**Comment by TingPing** [comment]
bot, build io.github.martinrotter.textosaurus

**Comment by martinrotter** [comment]
Thanks. :)

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/io.github.martinrotter.textosaurus

**Pull request closed without merging**